feat(coral): Topic Requests: add Toggle for "My requests" #777

roope-kar opened the pull request:
Makes it possible to view only the requests made by the current user

Resolves: #777

![image](https://user-images.githubusercontent.com/15416578/225542285-aa30a901-3316-4c3f-924e-10669d16e808.png)

programmiri commented on coral/src/app/features/components/table-filters/MyRequestFilter.tsx:
```diff
@@ -0,0 +1,31 @@
+      onChange={handleChangeIsMyRequest}
+      caption="Only show my requests"
+    >
+      Show My Requests
```
We could improve usability for screen reader (at least some) a bit if we wrap the "Show my requests" in an `<span aria-hidden="true">` (if that's possible).

Currently, VO reads: "Show my requests. Only show my requests". Which is a bit redundant. If the span works, it would only be "Only show my requests". 

🤔  On a second thought, maybe it would be even better to not have a caption but have "Only show my requests" as the main text. The caption is only one additional word to the main text, so it does not really add additional value. But that would have to be synced with Design of course.

roope-kar replied:
Really good points, thank you. I agree the caption feels redundant. I'll check with Mustafa if we can drop it. Otherwise i'll do the accessibility change you suggested 👍🏼

programmiri commented on coral/src/app/features/requests/components/topics/TopicRequests.tsx:
```diff
@@ -4,23 +4,29 @@ import { getTopicRequests } from "src/domain/topic/topic-api";
-      filters={[<TopicFilter key={"topic"} />]}
+      filters={[
+        <TopicFilter key={"topic"} />,
+        <MyRequestFilter key={"isMyRequest"} />,
```
I would add a test in for `TopicRequest` for the switch functionality, too. It will confirm that

- this component uses the `TopicSwitch`
- user using the switch triggers the correct behaviour -> api call with a changed payload. 

Both are the responsibility of this component.

roope-kar replied:
Sounds good! 👍🏼

roope-kar commented:
@programmiri Hey, committed changes and rebased.

The label was changed to _Only show my requests_. I think this is quite descriptive enough, so i did not add additional `aria-label`. 🤔 

Added test coverage also to `TopicRequests.tsx`

roope-kar commented:
Resolved

programmiri reviewed (approved):
Thanks for the additional tests! ✨
